src/toxcore/dht/codec.rs

```diff
@@ -132,9 +134,13 @@ impl Decoder for DhtCodec {
         }
 
         match Packet::from_bytes(buf) {
-            IResult::Incomplete(needed) => Err(DecodeError::incomplete_packet(needed, buf.to_vec())),
-            IResult::Error(error) => Err(DecodeError::deserialize(error, buf.to_vec())),
-            IResult::Done(_, packet) => {
+            Err(Err::Incomplete(needed)) => Err(DecodeError::incomplete_packet(needed, buf.to_vec())),
```

Comment:
Do we really need to handle all these variants? Why not just `Err(e) => ...`?

Reply:
How?

Reply:
Create one unified `DecodeError::Deserialize` variant and place there `nom::Err` rather than placing `Incomplete` and `Error` separately.

Reply:
Looks good

Reply:
Not fixed?